kedro_neptune/__init__.py

```diff
@@ -80,99 +107,130 @@ def init(metadata: ProjectMetadata, api_token: str, project: str, base_namespace
-class NeptuneMetadataDataSet(AbstractDataSet):
-    def __init__(self, run_id: str):
-        self._run_id: str = run_id
-        self._base_namespace: str = 'kedro'
-        self._run: Optional[neptune.Run] = None
+    if not context.catalog_file.exists():
+        with context.catalog_file.open("w") as catalog_file:
+            catalog_file.writelines(INITIAL_NEPTUNE_CATALOG)
 
-    def _save(self, data: Dict[str, Any]) -> None:
-        assert self._run is not None
 
-        self._run.assign(data)
+def get_neptune_config():
+    session: KedroSession = get_current_session()
+    context = session.load_context()
+    # pylint: disable=protected-access
+    credentials = context._get_config_credentials()
+    config = context.config_loader.get('neptune**')
+
+    api_token = credentials['neptune']['NEPTUNE_API_TOKEN']
+    project = config['neptune']['project']
+    base_namespace = config['neptune']['base_namespace'] or 'kedro'
+    source_files = config['neptune']['upload_source_files']
+
+    return api_token, project, base_namespace, source_files
+
+
+class NeptuneMetadataDataSet(AbstractDataSet):
+    def _save(self, data: Dict[str, Any]) -> None:
+        pass
 
     def _describe(self) -> Dict[str, Any]:
-        return getattr(self._run, 'get_structure', {})
+        pass
+
+    def _exists(self) -> bool:
+        return True
 
     def _load(self) -> neptune.run.Handler:
-        if not self._run:
-            session = get_current_session()
-            context = session.load_context()
-            # pylint: disable=protected-access
-            credentials = context._get_config_credentials()
-            config = context.config_loader.get('neptune**')
+        api_token, project, base_namespace, _ = get_neptune_config()
 
-            self._base_namespace = config['neptune']['base_namespace'] or 'kedro'
-            self._run = neptune.init(api_token=credentials['neptune']['NEPTUNE_API_TOKEN'],
-                                     project=config['neptune']['project'],
-                                     custom_run_id=self._run_id,
-                                     source_files=config['neptune']['upload_source_files'] or None)
+        run = neptune.init(api_token=api_token,
+                           project=project,
+                           capture_stdout=False,
+                           capture_stderr=False,
+                           capture_hardware_metrics=False,
+                           source_files=None)
 
-            self._run[INTEGRATION_VERSION_KEY] = __version__
+        return run[base_namespace]
 
-        return self._run[self._base_namespace]
 
+class BinaryFileDataSet(TextDataSet):
+    def __init__(
+            self,
+            filepath: str,
+            version: Version = None,
+            credentials: Dict[str, Any] = None,
+            fs_args: Dict[str, Any] = None,
+    ) -> None:
+        super(BinaryFileDataSet, self).__init__(
+            filepath=filepath,
+            version=None,
```

Comment:
maybe pass this `version` from params? ;)

Reply:
NeptuneFileDataSet is not versioned one (we cannot handle it so it's permamently set to None)